test/handlers/FranchiserFactoryHandler.sol

```diff
@@ -59,6 +59,33 @@ contract FranchiserFactoryHandler is Test {
         return bound(_amount, 0, 100_000_000e18);
     }
 
+    // Recursive function to get the sum tokens held in the sub-delegation tree of a given franchiser
+    function _getTotalAmountDelegatedByFranchiser(Franchiser _franchiser) internal returns (uint256 totalAmount) {
+        totalAmount = factory.votingToken().balanceOf(address(_franchiser));
+        for (uint256 i = 0; i < _franchiser.subDelegatees().length; i++) {
+            vm.startPrank(address(_franchiser));
+            Franchiser _subDelegatedFranchiser = _franchiser.getFranchiser(_franchiser.subDelegatees()[i]);
+            vm.stopPrank();
+            totalAmount +=  _getTotalAmountDelegatedByFranchiser(_subDelegatedFranchiser);
+        }
+    }
+
+    // Recursive function to get the selected subDelegatee of a given funded franchiser if possible
+    function _getSelectedSubDelegateeIfPossible(Franchiser _fundedFranchiser, uint256 _subDelegateIndex) internal returns (Franchiser _selectedFranchiser) {
```

Comment:
How does the index work for deeper levels of the tree?

Reply:
Renamed the function to `_getDeepestSubDelegateeInTree`.

Regarding index to the sub-delegates, it should use a number in the range of the sub-delegates array as the recursion descends the tree.  Console logging is showing, however, that it never seems to happen, mainly because the `...Many` function calls are creating so many Franchisers in the AddressSet that when a call to a `franchiser.subDelegate` function is triggered by the fuzz/invariant test runner, it seems to alway pick a franchiser that has no sub-delegates, thus there are no trees to be descended.. the tests are creating a lot of Franchisers with a single level of sub-delegation.  I think we'll need to add additional logic in the handler functions to ensure that we create some trees.. something like the handler for subdelegate automatically creates multiple levels of delegation.

Reply:
Ah okay that makes sense, maybe we have a handler that subdelegates on a subdelegate. Sounds good